src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -203,6 +214,39 @@ public void visitColumnWriteFragment(ColumnWriteFragment columnWriteFragment) {
     }
 
     // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Table Mutation: deletion
+
+    @Override
+    public void visitStandardTableDelete(TableDeleteStandard tableDeleteStandard) {
+        if (tableDeleteStandard.getWhereFragment() != null) {
+            // only applies when the table is a collection table with where restriction
+            throw new FeatureNotSupportedException();
+        }
```

Comment:
The above feature only applies for collection table (e.g. `one-to-many` or `many-to-many` association table), so it is far ahead of what we can verify (we need to finish table joining first, which might be the last task in the last milestone). Without verificaation, it is easy to implement it for the where fragment here is some native MQL filter.